File: 프로그래머스/2021 Dev-Matching: 웹 백엔드 개발자(상반기)/로또의 최고 순위와 최저 순위.py
# 로또의 최고 순위와 최저 순위
def solution(lottos, win_nums):
    right = 0
    for win in win_nums:
        if win in lottos:
            lottos.remove(win)
            right += 1
    possible = [1 for lotto in lottos if lotto == 0]
    row = 7 - right if right > 1 else 6
    high = 7 - (right + len(possible)) if right + len(possible) > 1 else 6
    return [high,row]

File: 프로그래머스/2021 Dev-Matching: 웹 백엔드 개발자(상반기)/로또의 최고 순위와 최저 순위.js
// 로또의 최고 순위와 최저 순위
function solution(lottos, win_nums) {
    let zero = 0, cnt = 0;
    for (let num of win_nums){
        if(lottos.includes(num))
            cnt += 1;
    }
    const zeroArr = lottos.filter(x => x === 0)
    const row = 7 - cnt < 6 ? 7 - cnt : 6;
    const high = 7 - (cnt + zeroArr.length) < 6 ? 7 - (cnt + zeroArr.length) : 6;
    return [high,row]
}